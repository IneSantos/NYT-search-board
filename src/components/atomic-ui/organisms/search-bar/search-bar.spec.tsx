import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import SearchBar from "./search-bar";

let store;
let mockedStore;

describe("SearchBar component tests", () => {
  test(`SearchBar component render`, () => {
    mockedStore = configureStore();
    store = mockedStore({
      article: {
        articleList: [
          {
            id: "",
            headline: "",
            abstract: "",
            keywords: [],
            section: "",
            snippet: "",
            pub_date: "",
            source: "",
            web_url: "",
          },
        ],
        page: 0,
        searchClicked: false,
      },
    });
    const { container } = render(
      <Provider store={store}>
        <SearchBar searchInput="This is an input" handleSearch={jest.fn()} />
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
