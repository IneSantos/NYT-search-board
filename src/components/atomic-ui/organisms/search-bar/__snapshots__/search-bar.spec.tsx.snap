// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component tests SearchBar component render 1`] = `
<div
  class="header-wrapper"
>
  <div
    class="search-wrapper"
  >
    <h2>
      Type search query term in here: 
    </h2>
    <div
      class="search"
    >
      <input
        class="search-term"
        placeholder="What are you looking for?"
        type="text"
        value="This is an input"
      />
      <button
        class="search-button"
        type="submit"
      >
        <i
          class="fas fa-search"
        />
      </button>
    </div>
  </div>
</div>
`;
